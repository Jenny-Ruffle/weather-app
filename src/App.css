.App {
  text-align: center;
  background-color: #7687a9;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-header{
  font-family: 'Jost', sans-serif;
  font-size: 46px;
  margin: 24px 0 12px;
}

.page-heading{
  font-family: 'Jost', sans-serif;
  font-size: 26px;
  margin: 0 0 48px 0;
}