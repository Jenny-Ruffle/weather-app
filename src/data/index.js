const data = [{
        day: 'Monday',
        date: '11/05',
        temp: '9',
        type: 'sunny',
        high: '19',
        low: '5'
    },
    {
        day: 'Tuesday',
        date: '12/05',
        temp: '12',
        type: 'rain',
        high: '19',
        low: '5'
    },
    {
        day: 'Wednesday',
        date: '13/05',
        temp: '14',
        type: 'rain',
        high: '19',
        low: '5'
    },
    {
        day: 'Thursday',
        date: '14/05',
        temp: '14',
        type: 'showers',
        high: '19',
        low: '5'
    },
    {
        day: 'Friday',
        date: '15/05',
        temp: '15',
        type: 'cloudy',
        high: '19',
        low: '5'
    },
    {
        day: 'Saturday',
        date: '16/05',
        temp: '13',
        type: 'cloudy',
        high: '19',
        low: '5'
    },
    {
        day: 'Sunday',
        date: '17/05',
        temp: '9',
        type: 'cloudy',
        high: '19',
        low: '5'
    },
    {
        day: 'Monday',
        date: '18/05',
        temp: '10',
        type: 'cloudy',
        high: '19',
        low: '5'
    },
    {
        day: 'Tuesday',
        date: '19/05',
        temp: '17',
        type: 'sunnyCloudy',
        high: '19',
        low: '5'
    },
    {
        day: 'Wednesday',
        date: '20/05',
        temp: '18',
        type: 'sunny',
        high: '19',
        low: '5'
    },
    {
        day: 'Thursday',
        date: '21/05',
        temp: '22',
        type: 'sunny',
        high: '19',
        low: '5'
    },
    {
        day: 'Friday',
        date: '22/05',
        temp: '20',
        type: 'snowy',
        high: '19',
        low: '5'
    },
    {
        day: 'Saturday',
        date: '23/05',
        temp: '17',
        type: 'hail',
        high: '19',
        low: '5'
    },
    {
        day: 'Sunday',
        date: '24/05',
        temp: '12',
        type: 'showers',
        high: '19',
        low: '5'
    },
    {
        day: 'Monday',
        date: '25/05',
        temp: '9',
        type: 'rain',
        high: '19',
        low: '5'
    }
]
export default data