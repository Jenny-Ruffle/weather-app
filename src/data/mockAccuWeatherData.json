
{
  "Headline": {
    "EffectiveDate": "2020-05-11T20:00:00+01:00",
    "EffectiveEpochDate": 1589223600,
    "Severity": 7,
    "Text": "Cool Monday night",
    "Category": "cold",
    "EndDate": "2020-05-12T08:00:00+01:00",
    "EndEpochDate": 1589266800,
    "MobileLink": "http://m.accuweather.com/en/gb/london/ec4a-2/extended-weather-forecast/328328?unit=c&lang=en-us",
    "Link": "http://www.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?unit=c&lang=en-us"
  },
  "DailyForecasts": [
    {
      "Date": "2020-05-11T07:00:00+01:00",
      "EpochDate": 1589176800,
      "Temperature": {
        "Minimum": {
          "Value": 2.8,
          "Unit": "C",
          "UnitType": 17
        },
        "Maximum": {
          "Value": 12.8,
          "Unit": "C",
          "UnitType": 17
        }
      },
      "Day": {
        "Icon": 4,
        "IconPhrase": "Intermittent clouds",
        "HasPrecipitation": false
      },
      "Night": {
        "Icon": 35,
        "IconPhrase": "Partly cloudy",
        "HasPrecipitation": false
      },
      "Sources": [
        "AccuWeather"
      ],
      "MobileLink": "http://m.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=1&unit=c&lang=en-us",
      "Link": "http://www.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=1&unit=c&lang=en-us"
    },
    {
      "Date": "2020-05-12T07:00:00+01:00",
      "EpochDate": 1589263200,
      "Temperature": {
        "Minimum": {
          "Value": 5,
          "Unit": "C",
          "UnitType": 17
        },
        "Maximum": {
          "Value": 13.3,
          "Unit": "C",
          "UnitType": 17
        }
      },
      "Day": {
        "Icon": 3,
        "IconPhrase": "Partly sunny",
        "HasPrecipitation": false
      },
      "Night": {
        "Icon": 35,
        "IconPhrase": "Partly cloudy",
        "HasPrecipitation": false
      },
      "Sources": [
        "AccuWeather"
      ],
      "MobileLink": "http://m.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=2&unit=c&lang=en-us",
      "Link": "http://www.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=2&unit=c&lang=en-us"
    },
    {
      "Date": "2020-05-13T07:00:00+01:00",
      "EpochDate": 1589349600,
      "Temperature": {
        "Minimum": {
          "Value": 4.4,
          "Unit": "C",
          "UnitType": 17
        },
        "Maximum": {
          "Value": 13.3,
          "Unit": "C",
          "UnitType": 17
        }
      },
      "Day": {
        "Icon": 4,
        "IconPhrase": "Intermittent clouds",
        "HasPrecipitation": false
      },
      "Night": {
        "Icon": 35,
        "IconPhrase": "Partly cloudy",
        "HasPrecipitation": false
      },
      "Sources": [
        "AccuWeather"
      ],
      "MobileLink": "http://m.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=3&unit=c&lang=en-us",
      "Link": "http://www.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=3&unit=c&lang=en-us"
    },
    {
      "Date": "2020-05-14T07:00:00+01:00",
      "EpochDate": 1589436000,
      "Temperature": {
        "Minimum": {
          "Value": 4.2,
          "Unit": "C",
          "UnitType": 17
        },
        "Maximum": {
          "Value": 13.4,
          "Unit": "C",
          "UnitType": 17
        }
      },
      "Day": {
        "Icon": 4,
        "IconPhrase": "Intermittent clouds",
        "HasPrecipitation": false
      },
      "Night": {
        "Icon": 33,
        "IconPhrase": "Clear",
        "HasPrecipitation": false
      },
      "Sources": [
        "AccuWeather"
      ],
      "MobileLink": "http://m.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=4&unit=c&lang=en-us",
      "Link": "http://www.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=4&unit=c&lang=en-us"
    },
    {
      "Date": "2020-05-15T07:00:00+01:00",
      "EpochDate": 1589522400,
      "Temperature": {
        "Minimum": {
          "Value": 6.3,
          "Unit": "C",
          "UnitType": 17
        },
        "Maximum": {
          "Value": 15.9,
          "Unit": "C",
          "UnitType": 17
        }
      },
      "Day": {
        "Icon": 3,
        "IconPhrase": "Partly sunny",
        "HasPrecipitation": false
      },
      "Night": {
        "Icon": 34,
        "IconPhrase": "Mostly clear",
        "HasPrecipitation": false
      },
      "Sources": [
        "AccuWeather"
      ],
      "MobileLink": "http://m.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=5&unit=c&lang=en-us",
      "Link": "http://www.accuweather.com/en/gb/london/ec4a-2/daily-weather-forecast/328328?day=5&unit=c&lang=en-us"
    }
  ]
}