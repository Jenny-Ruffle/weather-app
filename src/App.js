import React from 'react';
import {
  BrowserRouter as Router,
  Route
} from "react-router-dom";
import './App.css';
import FourteenDayForecast from './pages/FourteenDay'
import FiveDayForecast from './pages/FiveDay'
import ThreeDayForecast from './pages/ThreeDay'

function App() {
  return (
    <Router>
      <div className="App">
        <h1 className="App-header">
          Weather Forecast App
        </h1>
        <Route exact path="/" component={FiveDayForecast} />
        <Route exact path="/three-day" component={ThreeDayForecast} />
        <Route exact path="/fourteen-day" component={FourteenDayForecast} />
      </div>  
    </Router>
  );
}

export default App;

