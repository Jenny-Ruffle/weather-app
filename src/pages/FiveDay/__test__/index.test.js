import React from 'react';
import TestRenderer from 'react-test-renderer';
import FiveDayPage from '../index.js';
import data from '../../../data/mockAccuWeatherData.json'
import { ForecastContext } from "../../../App";

test("5 day forecast page render", () => {
    const newSnapshot = new TestRenderer.create(
        <ForecastContext.Provider value={data}>
            <FiveDayPage />
        </ForecastContext.Provider>
    ).toJSON();
    expect(newSnapshot).toMatchSnapshot();
});
