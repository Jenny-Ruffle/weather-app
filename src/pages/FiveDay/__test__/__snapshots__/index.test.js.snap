// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`5 day forecast page render 1`] = `
<div>
  <h3
    className="page-heading"
  >
    5 day forecast
  </h3>
  <div
    className="flex-carousel_wrapper"
  >
    <div
      className="flex-carousel"
    >
      <div
        className="weather-card"
      >
        <p
          className="weather-card_prefix"
        >
          11 / 5
        </p>
        <p
          className="weather-card_prefix"
        >
          Monday
        </p>
        <h1
          className="weather-card_temp"
        >
          8
          °c
        </h1>
        <img
          alt="weather-svg"
          src="sun-cloud.svg"
        />
        <p
          className="weather-card_prefix"
        >
          High: 
          13
          °c | Low: 
          3
          °c
        </p>
      </div>
      <div
        className="weather-card"
      >
        <p
          className="weather-card_prefix"
        >
          12 / 5
        </p>
        <p
          className="weather-card_prefix"
        >
          Tuesday
        </p>
        <h1
          className="weather-card_temp"
        >
          9
          °c
        </h1>
        <img
          alt="weather-svg"
          src="day.svg"
        />
        <p
          className="weather-card_prefix"
        >
          High: 
          13
          °c | Low: 
          5
          °c
        </p>
      </div>
      <div
        className="weather-card"
      >
        <p
          className="weather-card_prefix"
        >
          13 / 5
        </p>
        <p
          className="weather-card_prefix"
        >
          Wednesday
        </p>
        <h1
          className="weather-card_temp"
        >
          9
          °c
        </h1>
        <img
          alt="weather-svg"
          src="sun-cloud.svg"
        />
        <p
          className="weather-card_prefix"
        >
          High: 
          13
          °c | Low: 
          4
          °c
        </p>
      </div>
      <div
        className="weather-card"
      >
        <p
          className="weather-card_prefix"
        >
          14 / 5
        </p>
        <p
          className="weather-card_prefix"
        >
          Thursday
        </p>
        <h1
          className="weather-card_temp"
        >
          9
          °c
        </h1>
        <img
          alt="weather-svg"
          src="sun-cloud.svg"
        />
        <p
          className="weather-card_prefix"
        >
          High: 
          13
          °c | Low: 
          4
          °c
        </p>
      </div>
      <div
        className="weather-card"
      >
        <p
          className="weather-card_prefix"
        >
          15 / 5
        </p>
        <p
          className="weather-card_prefix"
        >
          Friday
        </p>
        <h1
          className="weather-card_temp"
        >
          11
          °c
        </h1>
        <img
          alt="weather-svg"
          src="day.svg"
        />
        <p
          className="weather-card_prefix"
        >
          High: 
          16
          °c | Low: 
          6
          °c
        </p>
      </div>
    </div>
  </div>
</div>
`;
